pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -563,7 +563,27 @@ func (d *DependenciesDistributor) removeScheduleResultFromAttachedBindings(bindi
 		delete(attachedBindings[index].Labels, bindingLabelKey)
 		updatedSnapshot := deleteBindingFromSnapshot(bindingNamespace, bindingName, attachedBindings[index].Spec.RequiredBy)
 		attachedBindings[index].Spec.RequiredBy = updatedSnapshot
-		attachedBindings[index].Spec.PreserveResourcesOnDeletion = nil
+		// Only adjust strategy fields for dependency-generated bindings (Placement == nil).
+		if attachedBindings[index].Spec.Placement == nil {
+			// When no remaining referrers, clear strategy fields.
+			if len(updatedSnapshot) == 0 {
+				attachedBindings[index].Spec.ConflictResolution = ""
+				attachedBindings[index].Spec.PreserveResourcesOnDeletion = nil
+			} else {
+				// Recompute effective strategy from remaining RequiredBy.
+				rbs, err := d.resolveResourceBindingFromSnapshots(context.TODO(), updatedSnapshot)
+				if err != nil {
+					klog.Errorf("Failed to resolve for resourceBinding(%s/%s): %v", binding.Namespace, binding.Name, err)
+					return err
```

Comment:
It is recommended to use the append err method here and continue processing other binding objects using continue.